app/src/main/kotlin/com/intive/picover/profile/viewmodel/ProfileViewModel.kt

```diff
@@ -0,0 +1,24 @@
+package com.intive.picover.profile.viewmodel
+
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.intive.picover.auth.repository.AuthRepository
+import dagger.hilt.android.lifecycle.HiltViewModel
+import javax.inject.Inject
+import kotlinx.coroutines.launch
+
+@HiltViewModel
+class ProfileViewModel @Inject constructor(
+	private val authRepository: AuthRepository,
+) : ViewModel() {
+
+	fun onLogoutClick() {
+		authRepository.logout()
+	}
+
+	fun onDeleteAccountClick() {
+		viewModelScope.launch {
+			authRepository.deleteAccount()
+		}
```

Comment:
what happened if some error occurs? for example there will be no internet connection? Shouldn't be wrapped in `runCatching`?

Reply:
The app will crash 😄 for now I've not added any error handling as you can see. I decided to skip it for now, and it will be a part of another task.

Reply:
Currently, that is only working for a happy-path but I think that's enough for now. I've created one issue #100 which also needs to be handled because account deletion is only possible with "fresh login credentials" so if you are logged for example for 2 days and try to delete your account the app will crash because according to Firebase we should first redirect the user to log-in again to make sure that the credentials are fresh and the user is 100% sure that he wanted to delete an account.